server/util/util.go

```diff
@@ -0,0 +1,99 @@
+package util
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"net/http"
+	"net/http/httptest"
+)
+
+const (
+	// ContentTypeJSON string associated with content-type json
+	ContentTypeJSON ContentType = "application/json"
+
+	// ContentTypeHTML string associated with content-type html
+	ContentTypeHTML ContentType = "text/html"
+
+	// ContentTypePlain
+	ContentTypePlain ContentType = "text/plain; charset=utf-8"
+)
+
+// ContentType type for http Content-Type
+type ContentType string
+
+// Request stores http Request basic data
+type Request struct {
+	Method string
+	URL    string
+	Body   interface{}
+}
+
+// ExpectedResponse stores expected responce basic data
+type ExpectedResponse struct {
+	StatusCode   int
+	ResponseType ContentType
+	Body         interface{}
+}
+
+// HTTPTest encapsulates data for testing needs
+type HTTPTest struct {
+	*assert.Assertions
```

Comment:
I leaning towards just passing `*testing.T` here, which would be more generic. Thoughts?